packages/node/src/integrations/tracing/fastify.ts

```diff
@@ -61,14 +61,27 @@ const _fastifyIntegration = (() => {
  */
 export const fastifyIntegration = defineIntegration(_fastifyIntegration);
 
+interface FastifyHandlerOptions {
+  /**
+   * Callback method deciding whether error should be captured and sent to Sentry
+   * @param error Captured middleware error
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  shouldHandleError?(this: void, request: any, reply: any, error: Error): boolean;
+}
+
 /**
  * Setup an error handler for Fastify.
  */
-export function setupFastifyErrorHandler(fastify: Fastify): void {
+export function setupFastifyErrorHandler(fastify: Fastify, options?: FastifyHandlerOptions): void {
   const plugin = Object.assign(
     function (fastify: Fastify, _options: unknown, done: () => void): void {
-      fastify.addHook('onError', async (_request, _reply, error) => {
-        captureException(error);
+      const shouldHandleError = options?.shouldHandleError || defaultShouldHandleError;
+
+      fastify.addHook('onError', async (request, reply, error) => {
```

Comment:
I think, since digging into this more, that my initial suggestion was incomplete - `reply.status` is probably going to be 200 here, and instead there is, I think, error.statusCode, according to the fastify-sentry plugin? https://github.com/immobiliare/fastify-sentry/blob/main/lib/base.js#L43-L46